test/server.locals.test.js

```diff
@@ -111,116 +125,259 @@ function cloneStack(stack) {
   return JSON.parse(JSON.stringify(stack));
 }
 
+function verifyThrownError(r) {
+  var addItemStub = r.addItemStub;
+
+  assert.isTrue(addItemStub.called);
+  var data = addItemStub.getCall(0).args[3].data;
+  assert.equal(data.body.trace_chain[0].exception.message, 'node error');
+  if (nodeMajorVersion < 10) {
+    // Node 8; locals disabled
+    var length = data.body.trace_chain[0].frames.length;
+    assert.equal(data.body.trace_chain[0].frames[length-1].locals, undefined);
+    assert.equal(data.body.trace_chain[0].frames[length-2].locals, undefined);
+  } else {
+    var length = data.body.trace_chain[0].frames.length;
+    assert.equal(data.body.trace_chain[0].frames[length-1].locals.error, '<Error object>');
+    assert.equal(data.body.trace_chain[0].frames[length-2].locals.timer, '<Timeout object>');
+  }
```

Comment:
This is similar comment to the one I made recently. IMHO it's better to stack versioning conditions from the most recent order.

```suggestion
  if (nodeMajorVersion >= 10) {
    // Node 10+; locals enabled
    var length = data.body.trace_chain[0].frames.length;
    assert.equal(data.body.trace_chain[0].frames[length-1].locals.error, '<Error object>');
    assert.equal(data.body.trace_chain[0].frames[length-2].locals.timer, '<Timeout object>');
  } else {
    // Node 8; locals disabled
    var length = data.body.trace_chain[0].frames.length;
    assert.equal(data.body.trace_chain[0].frames[length-1].locals, undefined);
    assert.equal(data.body.trace_chain[0].frames[length-2].locals, undefined);
  }
```